DXRMapFixups/DeusEx/Classes/DXRFixupVandenberg.uc

```diff
@@ -704,20 +725,69 @@ function TimerMapFixes()
         break;
 
     case "14_Oceanlab_silo":
-        prevMapsDone = dxr.flagbase.GetBool('Heliosborn') &&  //Finished Vandenberg, mission 12
-            dxr.flagbase.GetBool('schematic_downloaded'); //Finished Ocean Lab, mission 14,
-        prevMapsDone = prevMapsDone || !#defined(injections) || dxr.flags.settings.goals<=0;
+        // by design, no infolink plays after killing Howard Strong if the missile hasn't been redirected
+        _MissileLaunchedChecks();
+        _CheckMoreSiloInfolinks();
+        break;
+    }
+}
 
-        if(prevMapsDone
-           && dxr.flagbase.GetBool('missile_launched') //Redirected the missile in Silo, mission 14
-           && !dxr.flagbase.GetBool('MS_HowardStrongUnhidden'))
-        {
-            foreach AllActors(class'#var(prefix)HowardStrong', hs) {
-                hs.EnterWorld();
+function private _MissileLaunchedChecks() {
+    local #var(prefix)HowardStrong hs;
+    local bool strongAlive;
+    local BlackHelicopter chopper;
+
+    if (dxr.flagbase.GetBool('DXR_SiloEscapeHelicopterUnhidden') || !dxr.flagbase.GetBool('missile_launched')) {
+        return;
+    }
+
+    foreach AllActors(class'HowardStrong', hs) {
+        strongAlive = True;
+        break;
+    }
+
+    if (strongAlive == False) {
+        // both goals completed just now
+        foreach AllActors(class'BlackHelicopter', chopper, 'BlackHelicopter') {
+            chopper.EnterWorld();
+            break;
+        }
+        dxr.flagbase.SetBool('DXR_SiloEscapeHelicopterUnhidden', True,, 15);
+    } else {
+        // only computer goal completed
+        player().StartDataLinkTransmission("DL_Savage3");
+    }
+}
+
+function private _CheckMoreSiloInfolinks() {
+    local ConEvent ce;
+
+    if (!dxr.flagbase.GetBool('DXR_SiloEscapeHelicopterUnhidden')) {
+        return;
+    }
+
+    if (dxr.flagbase.GetBool('DL_Savage3_Played')) {
+        // both goals completed, computer infolink played
+        player().StartDataLinkTransmission("DL_Dead");
+    } else {
+        // both goals completed, computer infolink not played
+
+        for (ce = GetConversation('DL_Savage3').eventList; ce != None; ce = ce.nextEvent) {
+            if (ce.nextEvent != None && ce.nextEvent.eventType == ET_Speech) {
```

Comment:
making a local variable could save a bunch of these casts, something like

`local  ConEventSpeech nextSpeech`

```suggestion
            nextSpeech = ConEventSpeech(ce.nextEvent);
            if (nextSpeech  != None) {
```

Reply:
My preference is to not introduce new variables like that until it starts to harm readability otherwise. This way is one less step backward you need to keep in mind to read one of these lines.

Reply:
Vastly? It's three casts.